programs/keeper/Keeper.cpp

```diff
@@ -623,6 +624,19 @@ try
     buildLoggers(config(), logger());
     main_config_reloader->start();
 
+    std::optional<CgroupsMemoryUsageObserver> observer;
+    auto cgroups_memory_observer_wait_time = config().getUInt64("keeper_server.cgroups_memory_observer_wait_time", 1);
+    if (cgroups_memory_observer_wait_time > 0)
+    {
+        observer.emplace(std::chrono::seconds(cgroups_memory_observer_wait_time));
+        observer->startThread();
+        observer->setOnMemoryLimitUpdate([&]()
+        {
+            main_config_reloader->reload();
+        });
+    }
```

Comment:
The corresponding server code has a try/catch around the initialization for logging and error handling. Guess we should implement the same here.